api/v1/composition.go

```diff
@@ -214,3 +214,32 @@ func (s *CompositionStatus) GetCurrentSynthesisUUID() string {
 func (c *Composition) ShouldIgnoreSideEffects() bool {
 	return c.Annotations["eno.azure.io/ignore-side-effects"] == "true"
 }
+
+func (c *Composition) Synthesizing() bool {
+	return c.Status.CurrentSynthesis != nil && c.Status.CurrentSynthesis.Synthesized == nil
+}
+
+func (c *Composition) EnableIgnoreSideEffects() {
+	anno := c.GetAnnotations()
+	if anno == nil {
+		anno = map[string]string{}
+	}
+	anno["eno.azure.io/ignore-side-effects"] = "true"
+	c.SetAnnotations(anno)
+}
+
+const forceResynthesisAnnotation = "eno.azure.io/force-resynthesis"
```

Comment:
I'm pretty sure I'll get an answer after reading the rest of the code, but I'll leave this here in case the meaning is not clear later on.

Reply:
Currently we set the pending synthesis timestamp to force resynth if one of the slices is missing. Using this annotation allows us to target a specific synthesis. So if the states happened to be swapped concurrently we don't synthesize twice.

I also like the idea that it's an undocumented private API rather than being part of the status, but either way is probably fine here.